src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double MULTIPLIER_FOR_ARRAY_SIZE = 1.5;
+    private int arrayCapacity = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[arrayCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= arrayCapacity) {
+            newArray();
+        }
```

Comment:
Looks like duplicated code. Create an additional method for this code and call it `checkSize`, `checkCapacity`, or something like this and reuse it